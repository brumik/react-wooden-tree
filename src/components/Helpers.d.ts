export declare function defVal<T>(variable: T, defaultValue: T): T;
