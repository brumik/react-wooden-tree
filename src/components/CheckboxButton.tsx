import * as React from 'react';

/**
 * Callback function for CheckboxButton.
 */
export interface CheckboxButtonOnChange {
    (checked: boolean, id: string): void;
}

/**
 * CheckboxButton properties definition.
 */
interface CheckboxButtonProps {
    onChange: (checked: boolean) => void;
    checked: boolean;
    checkedIcon: string;
    partiallyCheckedIcon: string;
    uncheckedIcon: string;
}

/**
 * Creates a checkbox from button. On click event the callback function onChange is fired with the corresponding
 * value (if it was selected then with a false otherwise a true value is passed to the callback.)
 * Using fa-check-square, fa-square-o and fa-square for indicating the sates.
 *
 * @class CheckboxButton
 */
export class CheckboxButton extends React.Component<CheckboxButtonProps, {}> {
    render() {
        let icon: JSX.Element;
        let switchVal: boolean;

        switch (this.props.checked) {
            case false:
                icon = <i className={this.props.uncheckedIcon} />;
                switchVal = true;
                break;
            case undefined:
                icon = <i className={this.props.partiallyCheckedIcon} />;
                switchVal = true;
                break;
            case true:
                icon = <i className={this.props.checkedIcon} />;
                switchVal = false;
                break;
            default:
                icon = <i className="fa fa-question-circle" />;
        }

        return (
            <button className="SelectButton" onClick={() => this.props.onChange(switchVal)}>
                    {icon}
            </button>
        );
    }
}
