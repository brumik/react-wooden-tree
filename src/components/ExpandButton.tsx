import * as React from 'react';

export interface ExpandButtonProps {
    onChange: (checked: boolean) => void;
    expanded: boolean;
    loading: boolean; // null when error occurred.
    expandIcon: string;
    collapseIcon: string;
    loadingIcon: string;
    errorIcon: string;
}

export interface ExpandButtonOnChange {
    (nodeId: string, opened: boolean): void;
}

export class ExpandButton extends React.Component<ExpandButtonProps, {}> {
    render() {
        let icon: string;
        if ( this.props.loading ) {
            icon = this.props.loadingIcon;
        } else if ( this.props.loading === null ) {
            icon = this.props.errorIcon;
        } else if (this.props.expanded) {
            icon = this.props.collapseIcon;
        } else {
            icon = this.props.expandIcon;
        }

        let cName = icon + ' Icon ExpandButton';

        return (
            <i className={cName} onClick={() => this.props.onChange(!this.props.expanded)} />
        );
    }
}
