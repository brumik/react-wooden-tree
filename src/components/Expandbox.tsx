import * as React from "react";
import {FormEventHandler} from "react";


interface ExpandboxProps {
    onChange: FormEventHandler<HTMLInputElement>,
    expanded: boolean,
}

interface ExpandboxOnChange {
    (id: string, opened: boolean) : void
}

export interface ExpandboxData {
    expanded: boolean,
    onChange: ExpandboxOnChange
}

export function ExpandboxDataFactory(expanded: boolean, onChange: ExpandboxOnChange) : ExpandboxData {
    return {expanded: expanded, onChange: onChange}
}

export class Expandbox extends React.Component<ExpandboxProps, {}> {
    render() {
        return (
            <input type="checkbox"
                   checked={this.props.expanded}
                   onChange={this.props.onChange}
            />
        );
    }
}
