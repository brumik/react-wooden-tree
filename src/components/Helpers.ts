export function defaultIfNotExists(variable:any, defaultValue:any) {
    return variable ? variable : defaultValue;
}