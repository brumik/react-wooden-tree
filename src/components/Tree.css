li {
    list-style: none;
}
.NoOpenButton {
    margin-left: 18px;
}
.SelectButton {
    width: 14px;
    background: transparent;
    border: none;
    outline: none;
    margin: 2px;
    padding: 0;
    cursor: pointer;
}
.ExpandButton {
    width: 14px;
    background: transparent;
    border: none;
    outline: none;
    margin: 2px;
    padding: 0;
}
.NodeIconImage {
    width: 14px;
    height: auto;
    margin: 0 4px -1px;
}