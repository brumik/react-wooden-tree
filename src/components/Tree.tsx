import * as React from "react";
import { ListProps, List } from "./List";

export interface TreeProps {
    id: string
    list?: ListProps
}

interface TreeState {
}

export class Tree extends React.Component<TreeProps, TreeState> {
    constructor(props: TreeProps) {
        super(props);
    }

    render() {
        return (
            <div id={this.props.id}>
                <p>Will be a tree</p>
                <List
                    id={this.props.id}
                    {...this.props.list}
                />
            </div>
        );
    }
}