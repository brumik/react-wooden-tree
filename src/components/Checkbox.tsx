import * as React from "react";
import {FormEventHandler} from "react";
import {defVal} from "./Helpers";

export interface CheckboxOnChange {
    (checked: boolean, id : string) : void
}

export interface CheckboxData {
    visible: boolean,
    checked: boolean,
    onChange: CheckboxOnChange,
    childrenCheckedCount: number,
}

export function CheckboxDataFactory(checkbox : CheckboxData, onChange : CheckboxOnChange) : CheckboxData {
    if ( checkbox != null )
        return {
            visible: defVal(checkbox.visible, false),
            checked: defVal(checkbox.checked, false),
            onChange: onChange,
            childrenCheckedCount: 0
        };
    else return {visible: false, checked: false, onChange: onChange, childrenCheckedCount: 0};
}

interface CheckboxProps {
    onChange: FormEventHandler<HTMLInputElement>
    checked: boolean
}

export class Checkbox extends React.Component<CheckboxProps, {}> {
    render() {
        return (
            <input type="checkbox"
                   checked={this.props.checked}
                   onChange={this.props.onChange}
            />
        )
    }
}