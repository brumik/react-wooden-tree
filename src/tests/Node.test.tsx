import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { Node, Tree } from '..';
import { ParentDataType } from '..';
import { ParentDataContext } from '../components/Tree';

let parentData: ParentDataType;
const initState = {
    expanded: false,
    selected: false,
    disabled: false,
    checked: false
};

const subNodes = [
    {text: 'Sub Node 0', nodeId: '0.0'},
    {text: 'Sub Node 1', nodeId: '0.1'}
];

const subNodesList = ['0.0', '0.1'];

beforeEach(() => {
    parentData = {
        // Non-redux
        tree: Tree.convertHierarchicalTree(subNodes),

        // Callbacks
        checkboxOnChange: null,
        expandOnChange: null,
        selectOnChange: null,
        onLazyLoad: null,
        showCheckbox: true,
        initSelectedNode: function(nodeId: string) { return nodeId; },

        // Icons
        showIcon: true,
        showImage: true,
        nodeIcon: 'fa fa-ban fa-fw',
        checkedIcon: 'fa fa-check-square',
        uncheckedIcon: 'fa fa-square-o',
        partiallyCheckedIcon: 'fa fa-square',
        collapseIcon: 'fa fa-angle-down',
        expandIcon: 'fa fa-angle-right',
        loadingIcon: 'fa fa-spinner fa-spin',
        errorIcon: 'fa-exclamation-triangle',
        selectedIcon: 'fa fa-check',

        // Styling
        changedCheckboxClass: 'changed-checkbox',
        selectedClass: 'selected',

        // Other
        checkboxFirst: true,
        connectedNode: null,
    };
});

describe('node', () => {
    it('should render basic node', () => {
        const node = renderer
            .create(
                <ParentDataContext.Provider value={parentData}>
                    <Node
                        text={'Rendered Node'}
                        nodeId={'0'}
                        state={initState}
                    />
                </ParentDataContext.Provider>
            )
            .toJSON();
        expect(node).toMatchSnapshot();
    });

    it('should not render empty children', () => {
        const node = renderer
            .create(
                <ParentDataContext.Provider value={parentData}>
                    <Node
                        text={'Rendered Node'}
                        nodeId={'0'}
                        state={{...initState, expanded: true}}
                        nodes={[]}
                    />
                </ParentDataContext.Provider>
            )
            .toJSON();
        expect(node).toMatchSnapshot();
    });

    it('should render switched node and checkbox icon', () => {
        const node = renderer
            .create(
                <ParentDataContext.Provider value={{...parentData, checkboxFirst: false}}>
                    <Node
                        text={'Rendered Node'}
                        nodeId={'0'}
                        state={initState}
                    />
                </ParentDataContext.Provider>
            )
            .toJSON();
        expect(node).toMatchSnapshot();
    });

    it('should render expanded node with children', () => {
        const node = renderer
            .create(
                <ParentDataContext.Provider value={parentData}>
                    <Node
                        text={'Rendered Node'}
                        nodeId={'0'}
                        state={{...initState, expanded: true}}
                        nodes={subNodesList}
                    />
                </ParentDataContext.Provider>
            )
            .toJSON();
        expect(node).toMatchSnapshot();
    });

    it('should not render children nodes if collapsed', () => {
        const node = renderer
            .create(
                <ParentDataContext.Provider value={parentData}>
                    <Node
                        text={'Rendered Node'}
                        nodeId={'0'}
                        state={initState}
                        nodes={subNodesList}
                    />
                </ParentDataContext.Provider>
            )
            .toJSON();
        expect(node).toMatchSnapshot();
    });

    it('should render children nodes if expanded', () => {
        const node = renderer
            .create(
                <ParentDataContext.Provider value={parentData}>
                    <Node
                        text={'Rendered Node'}
                        nodeId={'0'}
                        state={{...initState, expanded: true}}
                        nodes={subNodesList}
                    />
                </ParentDataContext.Provider>
            )
            .toJSON();
        expect(node).toMatchSnapshot();
    });

    it('should display unique selected icon', () => {
        const node = renderer
            .create(
                <ParentDataContext.Provider value={parentData}>
                    <Node
                        text={'Rendered Node'}
                        nodeId={'0'}
                        state={{...initState, selected: true}}
                        selectedIcon={'icon icon-selected'}
                    />
                </ParentDataContext.Provider>
            )
            .toJSON();
        expect(node).toMatchSnapshot();
    });

    it('should not display any icon (checkbox, icon, expand icon)', () => {
        const node = renderer
            .create(
                <ParentDataContext.Provider value={{...parentData, showCheckbox: false, showIcon: false}}>
                    <Node
                        text={'Rendered Node'}
                        nodeId={'0'}
                        state={initState}
                        hideCheckbox={true}
                    />
                </ParentDataContext.Provider>
            )
            .toJSON();
        expect(node).toMatchSnapshot();
    });

    it('should display custom icon', () => {
        const node = renderer
            .create(
                <ParentDataContext.Provider value={{...parentData, showCheckbox: true, showIcon: true}}>
                    <Node
                        text={'Rendered Node'}
                        nodeId={'0'}
                        state={initState}
                        hideCheckbox={true}
                        icon={'fa fa-custom-icon'}
                    />
                </ParentDataContext.Provider>
            )
            .toJSON();
        expect(node).toMatchSnapshot();
    });

    it('should display custom image as icon', () => {
        const node = renderer
            .create(
                <ParentDataContext.Provider value={{...parentData, showCheckbox: true, showIcon: true}}>
                    <Node
                        text={'Rendered Node'}
                        nodeId={'0'}
                        state={initState}
                        hideCheckbox={true}
                        icon={'fa fa-custom-icon'}
                        image={'https://www.wpsuperstars.net/wp-content/uploads/2015/01/59.png'}
                    />
                </ParentDataContext.Provider>
            )
            .toJSON();
        expect(node).toMatchSnapshot();
    });

    it('should display loading icon', () => {
        const node = renderer
            .create(
                <ParentDataContext.Provider value={parentData}>
                    <Node
                        text={'Rendered Node'}
                        nodeId={'0'}
                        state={initState}
                        loading={true}
                        lazyLoad={true}
                    />
                </ParentDataContext.Provider>
            )
            .toJSON();
        expect(node).toMatchSnapshot();
    });

    it('should display error lazy loading icon', () => {
        const node = renderer
            .create(
                <ParentDataContext.Provider value={parentData}>
                    <Node
                        text={'Rendered Node'}
                        nodeId={'0'}
                        state={initState}
                        loading={null}
                        lazyLoad={true}
                    />
                </ParentDataContext.Provider>
            )
            .toJSON();
        expect(node).toMatchSnapshot();
    });

    it('should add custom class', () => {
        const node = renderer
            .create(
                <ParentDataContext.Provider value={parentData}>
                    <Node
                        text={'Rendered Node'}
                        nodeId={'0'}
                        state={initState}
                        classes={'custom-class'}
                    />
                </ParentDataContext.Provider>
            )
            .toJSON();
        expect(node).toMatchSnapshot();
    });

    it('should indicate selected node (icon and class)', () => {
        const node = renderer
            .create(
                <ParentDataContext.Provider value={parentData}>
                    <Node
                        text={'Rendered Node'}
                        nodeId={'0'}
                        state={{...initState, selected: true}}
                    />
                </ParentDataContext.Provider>
            )
            .toJSON();
        expect(node).toMatchSnapshot();
    });

    it('should display partially checked icon', () => {
        const node = renderer
            .create(
                <ParentDataContext.Provider value={parentData}>
                    <Node
                        text={'Rendered Node'}
                        nodeId={'0'}
                        state={{...initState, checked: null}}
                    />
                </ParentDataContext.Provider>
            )
            .toJSON();
        expect(node).toMatchSnapshot();
    });

    it('should display warning icon when undefined', () => {
        const node = renderer
            .create(
                <ParentDataContext.Provider value={parentData}>
                    <Node
                        text={'Rendered Node'}
                        nodeId={'0'}
                        state={{...initState, checked: undefined}}
                    />
                </ParentDataContext.Provider>
            )
            .toJSON();
        expect(node).toMatchSnapshot();
    });

    it('should indicate changed checked node (icon and class)', () => {
        const node = renderer
            .create(
                <ParentDataContext.Provider value={parentData}>
                    <Node
                        text={'Rendered Node'}
                        nodeId={'0'}
                        state={{...initState, checked: false}}
                    />
                </ParentDataContext.Provider>
            );

        // Simulate props change
        node.update(
            <ParentDataContext.Provider value={parentData}>
                <Node
                    text={'Rendered Node'}
                    nodeId={'0'}
                    state={{...initState, checked: true}}
                />
            </ParentDataContext.Provider>
        );

        expect(node.toJSON()).toMatchSnapshot();
    });

    it('should color the node icon with hex color', () => {
        const node = renderer
            .create(
                <ParentDataContext.Provider value={parentData}>
                    <Node
                        text={'Rendered Node'}
                        nodeId={'0'}
                        state={initState}
                        icon={'fa fa-circle'}
                        iconColor={'#F5F5F5'}
                    />
                </ParentDataContext.Provider>
            );
        expect(node.toJSON()).toMatchSnapshot();
    });

    it('should recolor the node icon background with rgb color', () => {
        const node = renderer
            .create(
                <ParentDataContext.Provider value={parentData}>
                    <Node
                        text={'Rendered Node'}
                        nodeId={'0'}
                        state={initState}
                        icon={'fa fa-circle'}
                        iconBackground={'rgb(100,255,100)'}
                    />
                </ParentDataContext.Provider>
            );
        expect(node.toJSON()).toMatchSnapshot();
    });

    it('should recolor both node icon ad icon background with color names', () => {
        const node = renderer
            .create(
                <ParentDataContext.Provider value={parentData}>
                    <Node
                        text={'Rendered Node'}
                        nodeId={'0'}
                        state={initState}
                        icon={'fa fa-circle'}
                        iconColor={'red'}
                        iconBackground={'green'}
                    />
              </ParentDataContext.Provider>
            );
        expect(node.toJSON()).toMatchSnapshot();
    });

    it('should recolor the default icon color and background with rgba colors', () => {
        const node = renderer
            .create(
                <ParentDataContext.Provider value={parentData}>
                    <Node
                        text={'Rendered Node'}
                        nodeId={'0'}
                        state={initState}
                        iconColor={'rgba(10,10,10,0.5)'}
                        iconBackground={'rgba(90,60,90,1)'}
                    />
                </ParentDataContext.Provider>
            );
        expect(node.toJSON()).toMatchSnapshot();

    });
});
