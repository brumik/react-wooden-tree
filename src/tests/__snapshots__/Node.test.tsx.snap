// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`node should add custom class 1`] = `
<li
  className="indent-0 NoOpenButton Selectable custom-class"
  data-id="0"
>
  <i
    className="fa fa-square-o Icon CheckboxButton"
    onClick={[Function]}
  />
  <i
    className="Icon fa fa-ban fa-fw"
    style={Object {}}
  />
  <span
    onClick={[Function]}
  >
    Rendered Node
  </span>
</li>
`;

exports[`node should color the node icon with hex color 1`] = `
<li
  className="indent-0 NoOpenButton Selectable"
  data-id="0"
>
  <i
    className="fa fa-square-o Icon CheckboxButton"
    onClick={[Function]}
  />
  <i
    className="Icon fa fa-circle"
    style={
      Object {
        "color": "#F5F5F5",
      }
    }
  />
  <span
    onClick={[Function]}
  >
    Rendered Node
  </span>
</li>
`;

exports[`node should display custom icon 1`] = `
<li
  className="indent-0 NoOpenButton Selectable"
  data-id="0"
>
  <i
    className="Icon fa fa-custom-icon"
    style={Object {}}
  />
  <span
    onClick={[Function]}
  >
    Rendered Node
  </span>
</li>
`;

exports[`node should display custom image as icon 1`] = `
<li
  className="indent-0 NoOpenButton Selectable"
  data-id="0"
>
  <img
    className="NodeIconImage"
    src="https://www.wpsuperstars.net/wp-content/uploads/2015/01/59.png"
  />
  <span
    onClick={[Function]}
  >
    Rendered Node
  </span>
</li>
`;

exports[`node should display error lazy loading icon 1`] = `
<li
  className="indent-0 Selectable"
  data-id="0"
>
  <i
    className="fa-exclamation-triangle Icon ExpandButton"
    onClick={[Function]}
  />
  <i
    className="fa fa-square-o Icon CheckboxButton"
    onClick={[Function]}
  />
  <i
    className="Icon fa fa-ban fa-fw"
    style={Object {}}
  />
  <span
    onClick={[Function]}
  >
    Rendered Node
  </span>
</li>
`;

exports[`node should display loading icon 1`] = `
<li
  className="indent-0 Selectable"
  data-id="0"
>
  <i
    className="fa fa-spinner fa-spin Icon ExpandButton"
    onClick={[Function]}
  />
  <i
    className="fa fa-square-o Icon CheckboxButton"
    onClick={[Function]}
  />
  <i
    className="Icon fa fa-ban fa-fw"
    style={Object {}}
  />
  <span
    onClick={[Function]}
  >
    Rendered Node
  </span>
</li>
`;

exports[`node should display partially checked icon 1`] = `
<li
  className="indent-0 NoOpenButton Selectable"
  data-id="0"
>
  <i
    className="fa fa-square Icon CheckboxButton"
    onClick={[Function]}
  />
  <i
    className="Icon fa fa-ban fa-fw"
    style={Object {}}
  />
  <span
    onClick={[Function]}
  >
    Rendered Node
  </span>
</li>
`;

exports[`node should display unique selected icon 1`] = `
<li
  className="indent-0 NoOpenButton Selectable selected"
  data-id="0"
>
  <i
    className="fa fa-square-o Icon CheckboxButton"
    onClick={[Function]}
  />
  <i
    className="icon icon-selected"
  />
  <i
    className="Icon fa fa-ban fa-fw"
    style={Object {}}
  />
  <span
    onClick={[Function]}
  >
    Rendered Node
  </span>
</li>
`;

exports[`node should display warning icon when undefined 1`] = `
<li
  className="indent-0 NoOpenButton Selectable"
  data-id="0"
>
  <i
    className="fa fa-question-circle Icon CheckboxButton"
    onClick={[Function]}
  />
  <i
    className="Icon fa fa-ban fa-fw"
    style={Object {}}
  />
  <span
    onClick={[Function]}
  >
    Rendered Node
  </span>
</li>
`;

exports[`node should indicate changed checked node (icon and class) 1`] = `
<li
  className="indent-0 NoOpenButton Selectable changed-checkbox"
  data-id="0"
>
  <i
    className="fa fa-check-square Icon CheckboxButton"
    onClick={[Function]}
  />
  <i
    className="Icon fa fa-ban fa-fw"
    style={Object {}}
  />
  <span
    onClick={[Function]}
  >
    Rendered Node
  </span>
</li>
`;

exports[`node should indicate selected node (icon and class) 1`] = `
<li
  className="indent-0 NoOpenButton Selectable selected"
  data-id="0"
>
  <i
    className="fa fa-square-o Icon CheckboxButton"
    onClick={[Function]}
  />
  <i
    className="fa fa-check"
  />
  <i
    className="Icon fa fa-ban fa-fw"
    style={Object {}}
  />
  <span
    onClick={[Function]}
  >
    Rendered Node
  </span>
</li>
`;

exports[`node should not display any icon (checkbox, icon, expand icon) 1`] = `
<li
  className="indent-0 NoOpenButton Selectable"
  data-id="0"
>
  <span
    onClick={[Function]}
  >
    Rendered Node
  </span>
</li>
`;

exports[`node should not render children nodes if collapsed 1`] = `
<li
  className="indent-0 Selectable"
  data-id="0"
>
  <i
    className="fa fa-angle-right Icon ExpandButton"
    onClick={[Function]}
  />
  <i
    className="fa fa-square-o Icon CheckboxButton"
    onClick={[Function]}
  />
  <i
    className="Icon fa fa-ban fa-fw"
    style={Object {}}
  />
  <span
    onClick={[Function]}
  >
    Rendered Node
  </span>
</li>
`;

exports[`node should not render empty children 1`] = `
<li
  className="indent-0 NoOpenButton Selectable"
  data-id="0"
>
  <i
    className="fa fa-square-o Icon CheckboxButton"
    onClick={[Function]}
  />
  <i
    className="Icon fa fa-ban fa-fw"
    style={Object {}}
  />
  <span
    onClick={[Function]}
  >
    Rendered Node
  </span>
</li>
`;

exports[`node should recolor both node icon ad icon background with color names 1`] = `
<li
  className="indent-0 NoOpenButton Selectable"
  data-id="0"
>
  <i
    className="fa fa-square-o Icon CheckboxButton"
    onClick={[Function]}
  />
  <i
    className="Icon fa fa-circle"
    style={
      Object {
        "backgroundColor": "green",
        "color": "red",
      }
    }
  />
  <span
    onClick={[Function]}
  >
    Rendered Node
  </span>
</li>
`;

exports[`node should recolor the default icon color and background with rgba colors 1`] = `
<li
  className="indent-0 NoOpenButton Selectable"
  data-id="0"
>
  <i
    className="fa fa-square-o Icon CheckboxButton"
    onClick={[Function]}
  />
  <i
    className="Icon fa fa-ban fa-fw"
    style={
      Object {
        "backgroundColor": "rgba(90,60,90,1)",
        "color": "rgba(10,10,10,0.5)",
      }
    }
  />
  <span
    onClick={[Function]}
  >
    Rendered Node
  </span>
</li>
`;

exports[`node should recolor the node icon background with rgb color 1`] = `
<li
  className="indent-0 NoOpenButton Selectable"
  data-id="0"
>
  <i
    className="fa fa-square-o Icon CheckboxButton"
    onClick={[Function]}
  />
  <i
    className="Icon fa fa-circle"
    style={
      Object {
        "backgroundColor": "rgb(100,255,100)",
      }
    }
  />
  <span
    onClick={[Function]}
  >
    Rendered Node
  </span>
</li>
`;

exports[`node should render basic node 1`] = `
<li
  className="indent-0 NoOpenButton Selectable"
  data-id="0"
>
  <i
    className="fa fa-square-o Icon CheckboxButton"
    onClick={[Function]}
  />
  <i
    className="Icon fa fa-ban fa-fw"
    style={Object {}}
  />
  <span
    onClick={[Function]}
  >
    Rendered Node
  </span>
</li>
`;

exports[`node should render children nodes if expanded 1`] = `
Array [
  <li
    className="indent-0 Selectable"
    data-id="0"
>
    <i
        className="fa fa-angle-down Icon ExpandButton"
        onClick={[Function]}
    />
    <i
        className="fa fa-square-o Icon CheckboxButton"
        onClick={[Function]}
    />
    <i
        className="Icon fa fa-ban fa-fw"
        style={Object {}}
    />
    <span
        onClick={[Function]}
    >
        Rendered Node
    </span>
</li>,
  <li
    className="indent-1 NoOpenButton Selectable"
    data-id="0.0"
>
    <i
        className="fa fa-square-o Icon CheckboxButton"
        onClick={[Function]}
    />
    <i
        className="Icon fa fa-ban fa-fw"
        style={Object {}}
    />
    <span
        onClick={[Function]}
    >
        Sub Node 0
    </span>
</li>,
  <li
    className="indent-1 NoOpenButton Selectable"
    data-id="0.1"
>
    <i
        className="fa fa-square-o Icon CheckboxButton"
        onClick={[Function]}
    />
    <i
        className="Icon fa fa-ban fa-fw"
        style={Object {}}
    />
    <span
        onClick={[Function]}
    >
        Sub Node 1
    </span>
</li>,
]
`;

exports[`node should render expanded node with children 1`] = `
Array [
  <li
    className="indent-0 Selectable"
    data-id="0"
>
    <i
        className="fa fa-angle-down Icon ExpandButton"
        onClick={[Function]}
    />
    <i
        className="fa fa-square-o Icon CheckboxButton"
        onClick={[Function]}
    />
    <i
        className="Icon fa fa-ban fa-fw"
        style={Object {}}
    />
    <span
        onClick={[Function]}
    >
        Rendered Node
    </span>
</li>,
  <li
    className="indent-1 NoOpenButton Selectable"
    data-id="0.0"
>
    <i
        className="fa fa-square-o Icon CheckboxButton"
        onClick={[Function]}
    />
    <i
        className="Icon fa fa-ban fa-fw"
        style={Object {}}
    />
    <span
        onClick={[Function]}
    >
        Sub Node 0
    </span>
</li>,
  <li
    className="indent-1 NoOpenButton Selectable"
    data-id="0.1"
>
    <i
        className="fa fa-square-o Icon CheckboxButton"
        onClick={[Function]}
    />
    <i
        className="Icon fa fa-ban fa-fw"
        style={Object {}}
    />
    <span
        onClick={[Function]}
    >
        Sub Node 1
    </span>
</li>,
]
`;

exports[`node should render not render empty children 1`] = `
<li
  className="indent-0 NoOpenButton Selectable"
  data-id="0"
>
  <i
    className="fa fa-square-o Icon CheckboxButton"
    onClick={[Function]}
  />
  <i
    className="Icon fa fa-ban fa-fw"
    style={Object {}}
  />
  <span
    onClick={[Function]}
  >
    Rendered Node
  </span>
</li>
`;

exports[`node should render switched node and checkbox icon 1`] = `
<li
  className="indent-0 NoOpenButton Selectable"
  data-id="0"
>
  <i
    className="Icon fa fa-ban fa-fw"
    style={Object {}}
  />
  <i
    className="fa fa-square-o Icon CheckboxButton"
    onClick={[Function]}
  />
  <span
    onClick={[Function]}
  >
    Rendered Node
  </span>
</li>
`;
